src/DontMergeMeYet/Services/CryptoHelper.cs

```diff
@@ -9,120 +14,10 @@ static class CryptoHelper
-
-                elements = GetIntegerSize(binaryReader);
-                var paramD = binaryReader.ReadBytes(elements);
-
-                elements = GetIntegerSize(binaryReader);
-                var paramP = binaryReader.ReadBytes(elements);
-
-                elements = GetIntegerSize(binaryReader);
-                var paramQ = binaryReader.ReadBytes(elements);
-
-                elements = GetIntegerSize(binaryReader);
-                var paramDP = binaryReader.ReadBytes(elements);
-
-                elements = GetIntegerSize(binaryReader);
-                var paramDQ = binaryReader.ReadBytes(elements);
-
-                elements = GetIntegerSize(binaryReader);
-                var paramIQ = binaryReader.ReadBytes(elements);
-
-                EnsureLength(ref paramD, 256);
-                EnsureLength(ref paramDP, 128);
-                EnsureLength(ref paramDQ, 128);
-                EnsureLength(ref paramE, 3);
-                EnsureLength(ref paramIQ, 128);
-                EnsureLength(ref paramModulus, 256);
-                EnsureLength(ref paramP, 128);
-                EnsureLength(ref paramQ, 128);
-
-                var rsaParameters = new RSAParameters
-                {
-                    Modulus = paramModulus,
-                    Exponent = paramE,
-                    D = paramD,
-                    P = paramP,
-                    Q = paramQ,
-                    DP = paramDP,
-                    DQ = paramDQ,
-                    InverseQ = paramIQ
-                };
-
-                return rsaParameters;
-            }
-        }
-
-        private static int GetIntegerSize(BinaryReader binary)
-        {
-            var bt = binary.ReadByte();
-
-            if (bt != 0x02)
-                return 0;
-
-            bt = binary.ReadByte();
-
-            int count;
-            if (bt == 0x81)
-            {
-                count = binary.ReadByte();
-            }
-            else if (bt == 0x82)
-            {
-                var highbyte = binary.ReadByte();
-                var lowbyte = binary.ReadByte();
-                byte[] modint = { lowbyte, highbyte, 0x00, 0x00 };
-                count = BitConverter.ToInt32(modint, 0);
-            }
-            else
-            {
-                count = bt;
-            }
-
-            while (binary.ReadByte() == 0x00)
-                count -= 1;
-
-            binary.BaseStream.Seek(-1, SeekOrigin.Current);
-
-            return count;
-        }
-
-        private static void EnsureLength(ref byte[] data, int desiredLength)
-        {
-            if (data == null || data.Length >= desiredLength)
-                return;
-
-            int zeros = desiredLength - data.Length;
-
-            byte[] newData = new byte[desiredLength];
-            Array.Copy(data, 0, newData, zeros, data.Length);
-
-            data = newData;
+            var privateKeyString = Encoding.UTF8.GetString(privateKeyBytes);
```

Comment:
I don't understand this line. Why would the private key bytes represent a valid UTF-8 string? Anyway, this doesn't seem to work... `ReadObject` returns null.

Reply:
I merely wanted to represent the bytes in a string so I could pass it to the `StringReader` below. Perhaps there's a better way to decode the Base64 encoded string such that it returns me a string. But the [top hit](https://stackoverflow.com/questions/7134837/how-do-i-decode-a-base64-encoded-string) I found suggested to do it this way.

The other thing that I could clean up here is to use `var` instead of declaring the types for each of these variables